infrastructure/templates/public-api/components/alerts/baseMemoryPercentageAlert.bicep

```diff
@@ -0,0 +1,38 @@
+import { Severity } from 'types.bicep'
+
+@description('Names of the resources that these alerts are being applied to.')
+param resourceNames string[]
+
+@description('Names of the resources that these alerts are being applied to.')
+param resourceType string
+
+@description('Names of the resources that these alerts are being applied to.')
+param metricName string
```

Comment:
Similar to `baseCpuPercentageAlert.bicep`, we could create some memory-specific types that can constrain these two parameters.